### PR TITLE
Field Additions / Changes

### PR DESCRIPTION
**Pull request by HouseOfHufflepuff**
Added fields to support more robust game logic to the slot resource.
- Comparator will enable slots with serial number =, <, > a given value.
- Required will enable optional slots.

Changed field "Open" to "Active" per @nolag .

### CODE REVIEW

**Review by pickerflicker — DISMISSED** [review]
lgtm

File: dss-collection-nft/contracts/DSSCollection.cdc [hunk 1]
```diff
@@ -317,6 +334,15 @@ pub contract DSSCollection: NonFungibleToken {
         return false
     }
 
+    // Validate comparator of slot
+    //
+    pub fun validateComparator(comparator: String): Bool {
+        if comparator == ">" || comparator == ">" || comparator == "=" {
```
Comment (pickerflicker) [inline dss-collection-nft/contracts/DSSCollection.cdc:340]: typo: "<"

**Review by nolag — DISMISSED** [review]
LGTM, do you want to make the change in Eric's code too (since it's checked in) or just leave that for another PR for him?

File: dss-collection-nft/contracts/DSSCollection.cdc [hunk 2]
```diff
@@ -100,6 +104,8 @@ pub contract DSSCollection: NonFungibleToken {
                 self.id = slot.id
                 self.collectionGroupID = slot.collectionGroupID
                 self.logicalOperator = slot.logicalOperator
+                self.comparator = slot.comparator
```
Comment (nolag) [inline dss-collection-nft/contracts/DSSCollection.cdc:107]: Should this be on the item instead?

**Review by alilloig — APPROVED** [review]

File: dss-collection-nft/contracts/DSSCollection.cdc (hunk 2 above)
Comment (HouseOfHufflepuff) [reply dss-collection-nft/contracts/DSSCollection.cdc:107]: It works both places and there are trade-offs. On item is more flexible and you can do more with it. On slot it aligns with the other logic & keeps things simpler.

Went with the simpler approach as you could replicate some of the flexibility you gain by just having more slots.

**Review by pickerflicker — APPROVED** [review]
lgtm on moving comparator to item from slot

**Review by nolag — APPROVED** [review]

File: dss-collection-nft/contracts/DSSCollection.cdc (hunk 2 above)
Comment (HouseOfHufflepuff) [reply dss-collection-nft/contracts/DSSCollection.cdc]: Updated to item.

**Pull request merged**